### PR TITLE
HL-725 Handler Applications

### PR DESCRIPTION
**Pull request by mjturt**
## Description :sparkles:

Handler applications

Includes tickets HL-637, HL-689, HL-688, HL-725, HL-734

- New application as handler frontend and backend implementations
- Company search functionality frontend/backend, refactored YRTTI and Service Bus integrations
- Translation fix HL-734 which is not related to handler applications
- Sentry baggage header fix (CORS error)

New environment variables:

- `YRTTI_BASE_URL`
- `YRTTI_DISABLE`
- `YRTTI_SEARCH_LIMIT`
- `SERVICE_BUS_BASE_URL`
- `SERVICE_BUS_SEARCH_LIMIT`

`YRTTI_BASIC_INFO_PATH` and `SERVICE_BUS_INFO_PATH` env variables were removed.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2053.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2053.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe employer result is __failed__ for https://kesaseteli-pr2053.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2053.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2053.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**EMPLOYER branch is deployed to platta: https://kesaseteli-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://handler-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**YOUTH branch is deployed to platta: https://youth-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**API branch is deployed to platta: https://kesaseteli-pr2053.api.dev.hel.ninja/healthz** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
**HANDLER branch is deployed to platta: https://handler-pr2053.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://kesaseteli-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
## TestCafe result is __success__ for https://youth-pr2053.dev.hel.ninja :laughing::tada::tada::tada:

File: frontend/benefit/handler/src/components/newApplication/formContent/attachmentsList/AttachmentsList.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import { BenefitAttachment } from 'shared/types/attachment';
+
+import { useAttachmentsList } from './useAttachmentsList';
+
+export type AttachmentsListProps = {
+  attachmentType: ATTACHMENT_TYPES;
+  attachments?: BenefitAttachment[];
+  handleQuietSave?: () => void;
+  required?: boolean;
+  as?: 'div' | 'li';
```
Comment (sirtawast) [inline frontend/benefit/handler/src/components/newApplication/formContent/attachmentsList/AttachmentsList.tsx:14]: `as` rings a bell. Is there something customized in this file or could this have been moved to benefit/shared from applicant side?

File: frontend/benefit/handler/src/components/newApplication/formContent/companySection/deMinimisAid/DeMinimisAidForm.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      >
+        <Button
+          theme="coat"
+          disabled={
+            !(
+              formik.values.granter &&
+              formik.values.amount &&
+              formik.values.grantedAt
+            ) ||
+            !formik.isValid ||
```
Comment (sirtawast) [inline frontend/benefit/handler/src/components/newApplication/formContent/companySection/deMinimisAid/DeMinimisAidForm.tsx:129]: Isn't formik.isValid stronger than those listed above?

File: frontend/benefit/handler/src/components/newApplication/formContent/companySection/deMinimisAid/utils/validation.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+      })),
+    [DE_MINIMIS_AID_KEYS.AMOUNT]: Yup.number()
+      .transform((_value, originalValue) => getNumberValue(originalValue))
+      .required(VALIDATION_MESSAGE_KEYS.REQUIRED)
+      .typeError(VALIDATION_MESSAGE_KEYS.NUMBER_INVALID)
+      .min(0, (param) => ({
+        min: param.min,
+        key: VALIDATION_MESSAGE_KEYS.NUMBER_MIN,
+      })),
+    [DE_MINIMIS_AID_KEYS.GRANTED_AT]: Yup.string()
```
Comment (sirtawast) [inline frontend/benefit/handler/src/components/newApplication/formContent/companySection/deMinimisAid/utils/validation.ts:29]: Granted at not required in yup but is checked to be filled to get the submit button enabled 🤔

File: frontend/benefit/handler/src/components/newApplication/review/companySection/CompanySection.tsx [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+              </$SummaryTableHeader>
+            </$GridCell>
+            <$GridCell>
+              <$SummaryTableHeader>
+                {t(
+                  `${translationsBase}.fields.deMinimisAidGrantedAt.labelShort`
+                )}
+              </$SummaryTableHeader>
+            </$GridCell>
+            {data.deMinimisAidSet?.map((aid: DeMinimisAid) => (
```
Comment (sirtawast) [inline frontend/benefit/handler/src/components/newApplication/review/companySection/CompanySection.tsx:123]: Could destructure with {granter, grantedAt, amount}

File: frontend/benefit/handler/src/components/newApplication/review/deMinimisSection/DeMinimisSection.tsx [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            <$SummaryTableHeader>
+              {t(`${translationsBase}.fields.deMinimisAidAmount.label`)}
+            </$SummaryTableHeader>
+          </$GridCell>
+          <$GridCell>
+            <$SummaryTableHeader>
+              {t(`${translationsBase}.fields.deMinimisAidGrantedAt.labelShort`)}
+            </$SummaryTableHeader>
+          </$GridCell>
+          {data.deMinimisAidSet?.map((aid: DeMinimisAid) => (
```
Comment (sirtawast) [inline frontend/benefit/handler/src/components/newApplication/review/deMinimisSection/DeMinimisSection.tsx:47]: {granter, grantedAt, amount}

Comment (sirtawast) [reply frontend/benefit/handler/src/components/newApplication/review/deMinimisSection/DeMinimisSection.tsx:47]: Actually, the whole review is basically the same as in `frontend/benefit/handler/src/components/newApplication/review/companySection/CompanySection.tsx`

File: frontend/benefit/handler/src/components/applicationList/ApplicationList.tsx [hunk 6]
```diff
@@ -1,9 +1,14 @@
-import { APPLICATION_STATUSES } from 'benefit-shared/constants';
+import { allApplicationStatuses } from 'benefit/handler/pages';
```
[the remark this answers is not in the document]
Comment (sirtawast) [reply frontend/benefit/handler/src/components/applicationList/ApplicationList.tsx:1]: Haha it's my code actually, rebasing has been done squashed :)

**Comment by sirtawast** [comment]
Looks good! So much new code, could not review it in full detail but did my best. Does the code really divert so much that it's necessary to fork it from `applicant`?

**Comment by mjturt** [comment]
> Looks good! So much new code, could not review it in full detail but did my best. Does the code really divert so much that it's necessary to fork it from `applicant`?

There are definitely still opportunities for refactoring. The optimal solution would be to refactor the form components into something that can be used on both sides, and then put them under shared code. But for now, I think it's ok, as this has been a long task.

**Comment by sirtawast** [comment]
> > Looks good! So much new code, could not review it in full detail but did my best. Does the code really divert so much that it's necessary to fork it from `applicant`?
> 
> There are definitely still opportunities for refactoring. The optimal solution would be to refactor the form components into something that can be used on both sides, and then put them under shared code. But for now, I think it's ok, as this has been a long task.

Yes I agree, no need to clean up now. Would need some sort of testing done anyway.

File: backend/benefit/companies/api/v1/views.py [hunk 7]
```diff
@@ -3,48 +3,55 @@
 from django.conf import settings
 from django.db import transaction
 from django.http import HttpRequest
-from drf_spectacular.utils import extend_schema, OpenApiParameter, OpenApiTypes
+from django.utils.translation import gettext_lazy as _g
```
Comment (rikuke) [inline backend/benefit/companies/api/v1/views.py:6]: No biggie, but I think in other parts of the codebase the gettext_lazy is imported and used just as  a "_"

File: backend/benefit/companies/services.py [hunk 8]
```diff
@@ -27,33 +34,42 @@ def get_or_create_organisation_with_business_id(business_id: str) -> Company:
...
 
 
 def get_or_create_association_with_business_id(business_id: str) -> Company:
-    """
-    Create a company instance using the YTJ integration.
-    """
+    """Create a company instance using the Yrtti integration."""
+
     yrtti_client = YRTTIClient()
+    association_data = yrtti_client.get_association_info_with_business_id(business_id)
```
[the remark this answers is not in the document]
Comment (rikuke) [reply backend/benefit/companies/services.py:57]: Doesn't the `_post() `raise at least for  HTTP 404, 500 etc. as it calls `response.raise_for_status()` ?

File: backend/benefit/companies/api/v1/views.py (hunk 7 above)
Comment (mjturt) [reply backend/benefit/companies/api/v1/views.py:6]: `_` was already in the scope of some function where "defined but not used" -warning needed to be silenced by naming some function argument as `_`. I will see if there is another way to deal with this.

File: backend/benefit/companies/services.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (mjturt) [reply backend/benefit/companies/services.py:57]: Both integrations `_post()`:s raises with `.raise_for_status()`, just like @rikuke said.

File: frontend/benefit/handler/src/components/applicationList/ApplicationList.tsx [hunk 9]
```diff
@@ -18,6 +23,16 @@ export interface ApplicationListProps {
...
 }
 
+const buildApplicationUrl = (
+  id: string,
+  status: APPLICATION_STATUSES
+): string => {
+  if (status === APPLICATION_STATUSES.DRAFT) {
+    return `/new-application?id=${String(id)}`;
+  }
+  return `/application?id=${String(id)}`;
```
[the remark this answers is not in the document]
Comment (sirtawast) [reply frontend/benefit/handler/src/components/applicationList/ApplicationList.tsx:33]: Also use constant for /new-application and /application routes!

File: backend/benefit/companies/api/v1/views.py (hunk 7 above)
Comment (sirtawast) [reply backend/benefit/companies/api/v1/views.py:6]: > `_` was already in the scope of some function where "defined but not used" -warning needed to be silenced by naming some function argument as `_`. I will see if there is another way to deal with this.

Maybe `__` xD

**Review by rikuke — APPROVED** [review]
Looks good, nothing big to comment on.

File: frontend/benefit/handler/src/components/newApplication/formContent/companySection/deMinimisAid/DeMinimisAidForm.tsx (hunk 2 above)
Comment (mjturt) [reply frontend/benefit/handler/src/components/newApplication/formContent/companySection/deMinimisAid/DeMinimisAidForm.tsx:129]: Added this just from the applicant side. The whole deminimis things are 90% identical on both sides, and it feels illegal. Maybe refactor some day.

File: frontend/benefit/handler/src/components/newApplication/review/deMinimisSection/DeMinimisSection.tsx (hunk 5 above)
Comment (mjturt) [reply frontend/benefit/handler/src/components/newApplication/review/deMinimisSection/DeMinimisSection.tsx:47]: Good catch! The whole `DeMinimisSection` is completely useless.

**Review by sirtawast — APPROVED** [review]

**Pull request closed without merging**